tests/queries/0_stateless/03403_function_search_index.sql

```diff
@@ -0,0 +1,95 @@
+
+SELECT 'Skip index should choose 1 part and 1024 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'Hallo');
+
+SELECT 'Skip index should choose 1 part and 1024 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'Hallo Word'); -- Word does not exist in terms
+
+SELECT 'Skip index should choose 2 parts and 2048 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'Hello');
+
+SELECT 'Skip index should choose 2 parts and 2048 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'Hello World');
+
+SELECT 'Skip index should choose 2 parts and 2048 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'Hallo World');
+
+SELECT 'Skip index should choose 3 parts and 3072 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'Hello Hallo');
+
+SELECT 'Skip index should choose 3 parts and 3072 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'ClickHouse');
+
+SELECT 'Skip index should choose all 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAny(message, 'ClickHouse World');
+
+SELECT 'searchAll is used during index analysis';
+
+SELECT 'Skip index should choose none for non-existent term';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Click');
+
+SELECT 'Skip index should choose 1 part and 1024 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Hallo');
+
+SELECT 'Skip index should choose 1 part and 1024 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Hello World');
+
+SELECT 'Skip index should choose none if any term does not exists in dictionary';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Hallo Word'); -- Word does not exist in terms
+
+SELECT 'Skip index should choose 2 parts and 2048 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Hello');
+
+SELECT 'Skip index should choose none';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Hallo World');
+
+SELECT 'Skip index should choose none';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'Hello Hallo');
+
+SELECT 'Skip index should choose 3 parts and 3072 granules out of 4 parts and 4096 granules';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'ClickHouse');
+
+SELECT 'Skip index should choose none';
+EXPLAIN indexes=1
+SELECT count() FROM tab WHERE searchAll(message, 'ClickHouse World');
+
+DROP TABLE tab;
```

Comment:
As far as I understand, in above test, either all or none of the rows/granules are expected to match within each part. Maybe it makes sense to test the case that there are mixed matching/non-matching rows/granules within a part?